frontend/public/components/overview/deployment-overview.tsx

```diff
@@ -53,12 +55,26 @@ const tabs = [
   },
 ];
 
+const getResourceTabComp = (t) => (props) => <AsyncComponent loader={t.properties.loader} {...props} />;
+
+const getPluginTabResources = (item): any[] => {
+  const tabEntry = plugins.registry.getOverviewResourceTab().filter(tab => item[tab.properties.model.plural]);
+  tabEntry.map(t => {
```

Comment:
Don't modify component definition in tabs array. Instead construct a new object and return using map.
Also avoid nested map function by using Array.find to locate any specific configurations.